up42/tools.py

```diff
@@ -223,22 +229,33 @@ def get_block_details(self, block_id: str, as_dataframe: bool = False) -> dict:
         else:
             return details_json
 
+    @check_auth
+    def get_block_coverage(self, block_id: str) -> Dict[str, Optional[str]]:
+        # pylint: disable=unused-argument
+        try:
+            url = f"{self.auth._endpoint()}/blocks/{block_id}/coverage"  # public blocks
+            response_json = self.auth._request(request_type="GET", url=url)
+            details_json = response_json["data"]
+            return details_json
+        except Exception as e:  # pylint: disable=broad-except
```

Comment:
Yes, I think is an error that can be extracted from the requests.errors, I will check for it.

Reply:
Usually this should be handled by the auth module then already and surface the API error. Unless the API error message is very non-descriptive, then maybe makes sense to reraise here.

Reply:
OK. The thing is that if the block has not coverage and we don't add the exception it will return a 404 error. In this case, it will return a url=None with same structure as the usual response.

Reply:
Cool. Looked into the error raise, was also a bit confused because it is such a long ugly stac trace, but indeed currently in the SDK if a function raises a requests error we present that error to the user to not have custom error solutions in the SDK. Worth a discussion in the future though def.